dali/test/python/test_detection_pipeline.py

```diff
@@ -13,147 +13,342 @@
+
+            # Check cropping ops
+            decode_crop = compare(images_ssd_crop, images_decode_crop)
+            slice_cpu = compare(images_ssd_crop, images_slice_cpu)
+            slice_gpu = compare(images_ssd_crop, images_slice_gpu)
+            images_crop = decode_crop and slice_cpu and slice_gpu
+            boxes_crop = compare(boxes_ssd_crop, boxes_random_crop)
+            labels_crop = compare(labels_ssd_crop, labels_random_crop)
+            crop = images_crop and boxes_crop and labels_crop
+
+            # Check normalizing ops
+            normalize = compare(images_normalized_cpu, images_normalized_gpu)
+
+            # Check flipping ops
+            images_flip = compare(images_flipped_cpu, images_flipped_gpu)
+            boxes_flip = compare(boxes_flipped_cpu, boxes_flipped_gpu)
+            flip = images_flip and boxes_flip
+
+            # Check box encoding ops
+            encoded_boxes = compare(encoded_boxes_cpu, encoded_boxes_gpu)
+            encoded_labels = compare(encoded_labels_cpu, encoded_labels_gpu)
+            box_encoder = encoded_boxes and encoded_labels
+
+            if not crop or not normalize or not flip or not box_encoder:
+                print('Error during iteration', iter)
+                print('Crop = ', crop)
+                print('  decode_crop =', decode_crop)
+                print('  slice_cpu =', slice_cpu)
+                print('  slice_gpu =', slice_gpu)
+                print('  boxes_crop =', decode_crop)
+                print('  labels_cpu =', labels_crop)
+
+                print('Normalize =', normalize)
+
+                print('Flip =', flip)
+                print('  images_flip =', images_flip)
+                print('  boxes_flip =', boxes_flip)
+
+                print('Box encoder =', box_encoder)
+                print('  encoded_boxes =', encoded_boxes)
+                print('  encoded_labels =', encoded_labels)
+
+                exit(1)
+
+        if not iter % 100:
+            print("Iteration: {}/ {}".format(iter + 1, args.iters))
+
+    print()
+
+
+def print_args(args):
+    print('Args values:')
+    for arg in vars(args):
+        print('{0} = {1}'.format(arg, getattr(args, arg)))
+    print()
+
+
+def run_test(args):
+    print_args(args)
+
+    for dataset in data_paths():
+        print('Run DetectionPipeline test for', dataset[0])
+        run_for_dataset(args, dataset)
+
+
+def random_seed():
+    return int(random.random() * (1 << 32))
+
+
+def make_parser():
+    parser = argparse.ArgumentParser(description='Detection pipeline test')
+    parser.add_argument(
+        '-i', '--iters', default=None, type=int, metavar='N',
+        help='number of iterations to run (default: whole dataset)')
+    parser.add_argument(
+        '-g', '--num_gpus', default=1, type=int, metavar='N',
+        help='number of GPUs (default: %(default)s)')
+    parser.add_argument(
+        '-s', '--seed', default=random_seed(), type=int, metavar='N',
```

Comment:
Is seed printed somewhere?

Reply:
Yes, `print(args)` in line 328 prints all arguments